File: algorithms/Final Prices With a Special Discount in a Shop/Final Prices With a Special Discount in a Shop.java
class Solution {
    public int[] finalPrices(int[] prices) {
        for(int i = 0; i < prices.length; i++){
            for(int j = i + 1; j < prices.length; j++){
                if(j > i && prices[j] <= prices[i]){
                    prices[i] -= prices[j];
                    break;
                }
            }            
        }
        return prices;
    }
}

File: scripts/algorithms/F/Final Prices With a Special Discount in a Shop/Final Prices With a Special Discount in a Shop.js
var finalPrices = function(prices) {
    let adjPrices = [];
    while(prices.length) {
        let currentPrice = prices[0];
        
        // Remove first price in original array
        // Since we're looking for a price less than or equal to
        prices.shift();
        
        let lowerPrice = prices.find((a) => a <= currentPrice);
        adjPrices.push(lowerPrice ? currentPrice - lowerPrice : currentPrice);
    }
    
    return adjPrices;
};
